src/editor.rs

```diff
@@ -359,6 +359,49 @@ impl Editor {
         Ok(())
     }
 
+    //TODO: support multi-function
```

Comment:
Can you describe what would need to be changed here for multi-function? (at a high level)